### PR TITLE
[iOS-Sync] Delete Other Devices (uplift to 1.20.x)

### PR DESCRIPTION
**Pull request by bsclifton**
- Adds support for deleting other devices from the sync chain.
- Adds support for the current device being deleted from the chain, by another device.
- Fixes missing brave_services_key

Uplift of https://github.com/brave/brave-core/pull/7584
Resolves brave/brave-ios#3194 and brave/brave-ios#3195

Approved, please ensure that before merging: 
- [x] You have checked CI and the builds, lint, and tests all pass or are not related to your PR. 
- [ ] You have tested your change on Nightly. 
- [x] The PR milestones match the branch they are landing to. 

After you merge: 
- [x] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

**Comment by kjozwiak** [comment]
@bsclifton looks like there's a few `lint` issues as per https://ci.brave.com/blue/organizations/jenkins/pr-brave-browser-ios-1.20.x-sync-delete-devices-post-init/detail/pr-brave-browser-ios-1.20.x-sync-delete-devices-post-init/2/pipeline. Probably doesn't block uplift but should we create anew issue to address these?

```
[2021-02-01T17:43:34.149Z] Running cpplint...
[2021-02-01T17:43:34.149Z] Skipping file ios/browser/api/sync/BUILD.gn
[2021-02-01T17:43:34.149Z] Done processing ios/browser/api/sync/brave_sync_api.h
[2021-02-01T17:43:34.149Z] Skipping file ios/browser/api/sync/brave_sync_api.mm
[2021-02-01T17:43:34.405Z] ios/browser/api/sync/brave_sync_worker.cc:247:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
[2021-02-01T17:43:34.405Z] ios/browser/api/sync/brave_sync_worker.cc:251:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
[2021-02-01T17:43:34.405Z] ios/browser/api/sync/brave_sync_worker.cc:255:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
[2021-02-01T17:43:34.405Z] ios/browser/api/sync/brave_sync_worker.cc:264:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
[2021-02-01T17:43:34.405Z] ios/browser/api/sync/brave_sync_worker.cc:268:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
[2021-02-01T17:43:34.405Z] ios/browser/api/sync/brave_sync_worker.cc:272:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
[2021-02-01T17:43:34.405Z] Done processing ios/browser/api/sync/brave_sync_worker.cc
[2021-02-01T17:43:34.405Z] Done processing ios/browser/api/sync/brave_sync_worker.h
[2021-02-01T17:43:34.405Z] Skipping clang/gn format check because base_branch is origin/1.20.x instead of refs/remotes/origin/master

[2021-02-01T17:43:34.405Z] cpplint errors found: 6
```

**Comment by bsclifton** [comment]
whoops - forgot to comment here; pushed lint fixes 👍

**Comment by kjozwiak** [comment]
`test-browser` failed on `macOS` due to `RewardsPromotionBrowserTest.ClaimViaPanel` failing as per https://ci.brave.com/job/pr-brave-browser-ios-1.20.x-sync-delete-devices-macos/3/execution/node/222/log/. However, this is a known intermittent issue as per https://github.com/brave/brave-browser/issues/12605.

```
17:27:59  [  FAILED  ] RewardsPromotionBrowserTest.ClaimViaPanel, where TypeParam =  and GetParam() =  (31123 ms)
17:27:59  [669/669] RewardsPromotionBrowserTest.ClaimViaPanel (31418 ms)
17:27:59  1 test failed:
17:27:59      RewardsPromotionBrowserTest.ClaimViaPanel (../../brave/components/brave_rewards/browser/test/rewards_promotion_browsertest.cc:204)
```

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.20.x` approved after deliberating with @brave/uplift-approvers.

**Pull request merged**
